api/controllers/UserController.js

```diff
@@ -214,26 +215,168 @@ module.exports = {
+          });
+          var svexFlairPromise = new Promise(function(resolve, reject) {
+            Ban.giveBannedUserFlair(req.user.redToken, req.params.username, flair2.flair_css_class, flair2.flair_text, 'SVExchange', resolve, reject);
+          });
+          var ptradesAutomodPromise = new Promise(function(resolve, reject) {
+            Ban.updateAutomod(req.user.redToken, req.params.username, 'pokemontrades', unique_fcs, resolve, reject);
+          });
+          var svexAutomodPromise = new Promise(function(resolve, reject) {
+            Ban.updateAutomod(req.user.redToken, req.params.username, 'SVExchange', unique_fcs, resolve, reject);
+          });
+          var removeTSVPromise = new Promise(function(resolve, reject) {
+            Ban.removeTSVThreads(req.user.redToken, req.params.username, resolve, reject);
+          });
+          var updateBanlistPromise = new Promise(function(resolve, reject) {
+            Ban.updateBanlist(req.user.redToken, req.params.username, req.params.banlistEntry, unique_fcs, igns, resolve, reject);
+          });
+          var localBanPromise = new Promise(function(resolve, reject) {
+            Ban.localBanUser(req.params.username, resolve, reject);
+          });
+          promises = [ //Tasks for permabanning
+            ptradesBanPromise,
+            svexBanPromise,
+            ptradesFlairPromise,
+            svexFlairPromise,
+            ptradesAutomodPromise,
+            svexAutomodPromise,
+            removeTSVPromise,
+            updateBanlistPromise,
+            localBanPromise
+          ];
+        }
+        Promise.all(promises).then(function(result) {
+          res.json('ok', 200);
+        }, function(error) {
+          console.log(error);
+          res.json({error: error}, 500);
+        });
+      }, req.params.username);
     });
   },
 
+  // Sets a local ban on a user. There's no frontend, but an authorized mod can call it directly through a POST request from the JS console.
+  // io.socket.post("/mod/setlocalban", {username: "example_user", ban: true});
+  setLocalBan: function(req, res) {
+    if (!req.user.isMod) {
+      return res.json({errors: "Not a mod"}, 403);
+    }
+    if (req.allParams().ban !== true && req.allParams().ban !== false) {
+      //Requires an explicit statement of true or false, to prevent accidental unbans if the parameter is undefined.
+      return res.json({errors: "Unspecified ban state"}, 400);
+    }
+    User.update({name: req.allParams().username}, {banned: req.allParams().ban}, function (err, updated) {
+      if (err) {
+        return res.json({errors: "Internal server error"}, 500);
+      }
+      if (req.allParams().ban) {
+        console.log("Locally banned " + req.allParams().username);
+      } else {
+        console.log("Locally unbanned " + req.allParams().username);
+      }
+      return res.json({errors: ''}, 200);
+    });
+  },
+  
```

Comment:
I decided to keep the backend for this even though we removed the button for locally banning/unbanning people. We shouldn't have to use this too often, but it could come in handy if we unban someone after an appeal. (Without this, there's no easy way to remove a user from the local banlist.)